packages/flow-core/src/components/f-carousel/f-carousel.ts

```diff
@@ -0,0 +1,173 @@
+import { html, unsafeCSS } from "lit";
+import { customElement, property, query } from "lit/decorators.js";
+import eleStyle from "./f-carousel.scss";
+import { FRoot } from "../../mixins/components/f-root/f-root";
+import { FDiv } from "../f-div/f-div";
+import { FCarouselContent } from "../f-carousel-content/f-carousel-content";
+
+@customElement("f-carousel")
+export class FCarousel extends FRoot {
+	/**
+	 * css loaded from scss file
+	 */
+	static styles = [unsafeCSS(eleStyle), ...FDiv.styles];
+
+	/**
+	 * @attribute f-carousel can have 2 sizes.
+	 */
+	@property({ type: String, reflect: true })
+	size?: "small" | "medium" = "medium";
+
+	/**
+	 * @attribute f-carousel interval for autoplay.
+	 */
+	@property({ type: Number, reflect: true })
+	interval?: number = 5000;
+
+	/**
+	 * @attribute f-carousel autoplay
+	 */
+	@property({ type: Boolean, reflect: true })
+	autoplay?: boolean = false;
+
+	/**
+	 * @attribute query-selector for carousel wrapper.
+	 */
+	@query(".f-carousel")
+	carouselWrapper!: FDiv;
+
+	allCarouselContents!: NodeListOf<FCarouselContent>;
```

Comment:
remove ! because if we render f-carousel without content then it will through an error.

use `? `and check for Nullish value before using it or assign `null` by default